common/ManagementTestShared/Redesign/MockTestEnvironment.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+using System;
+
+namespace Azure.ResourceManager.TestFramework
+{
+    public class MockTestEnvironment : TestEnvironment
```

Comment:
You don't need a test environment for mock tests because they would never run live.

Reply:
That is true, but I need the Credential to be the mock credential even not in Playback mode

Reply:
Can you use `new MockCredential()`?